rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -943,8 +943,19 @@ func Qualify_Transaction_ZeroCoupon: <"Qualifies a product as having the Transac
     output:
         is_product boolean (1..1)
     set is_product:
-        economicTerms -> payout -> interestRatePayout -> paymentDates -> paymentFrequency -> periodMultiplier all = 1
-            and economicTerms -> payout -> interestRatePayout -> paymentDates -> paymentFrequency -> period all = PeriodExtendedEnum -> T
+        economicTerms -> payout -> interestRatePayout -> paymentDates -> paymentFrequency
+        filter item -> periodMultiplier = 1 and item -> period = PeriodExtendedEnum -> T then exists
+
+func Qualify_Transaction_ZeroCoupon_KnownAmount: <"Qualifies a product as having the Transaction classification Zero Coupon with a Known Amount. This category applies to a Zero Coupon Swap in which the fixed leg pays a known amount at maturity.">
+    inputs:
+        economicTerms EconomicTerms (1..1)
+    output:
+        is_product boolean (1..1)
+    set is_product:
+        Qualify_SubProduct_FixedFloat(economicTerms) = True and
+        Qualify_Transaction_ZeroCoupon(economicTerms) = True and
+        (economicTerms -> payout -> interestRatePayout
+            filter item -> priceQuantity exists and rateSpecification is absent and paymentDates -> paymentFrequency -> periodMultiplier = 1 and paymentDates -> paymentFrequency -> period = PeriodExtendedEnum -> T then exists)
```

Comment:
the logic should rather check that if rateSpecification is present it is a fixed rate. The peer review group indicated that a Zerocoupon_knownamount can also feature accruals but they have to be deterministic i.e. fixed rate.

Reply:
No, it doesn't include the fixed rate. That's the whole point about ZC with a Known Amount. There is no fixed rate in the fixed leg. The Known Amount is the result of the accrued fixed rate calculations so the fixed rate is not included as part of the product representation.